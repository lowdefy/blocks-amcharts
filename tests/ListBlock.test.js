/*
  Copyright 2020 Lowdefy, Inc

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
*/

import React from 'react';
import { mockBlock, runBlockSchemaTests, runRenderTests } from '@lowdefy/block-tools';
import { configure, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
configure({ adapter: new Adapter() });

import { ListBlock } from '../src';
import examples from '../demo/examples/ListBlock.yaml';
import meta from '../src/blocks/ListBlock/ListBlock.json';

jest.spyOn(window, 'alert').mockImplementation(() => {});

runRenderTests({ examples, Block: ListBlock, meta });
runBlockSchemaTests({ examples, meta });

const { before, methods, getProps } = mockBlock({ meta });
beforeEach(before);

it('test button clicks', () => {
  const block = {
    id: 'one',
    type: 'InputBlock',
    areas: {
      content: {
        blocks: [
          {
            id: 'a',
          },
        ],
      },
    },
  };
  const Shell = () => <ListBlock {...getProps(block)} methods={methods} />;
  const wrapper = mount(<Shell />);
  wrapper.find('[data-testid="one-pushItem"]').simulate('click');
  expect(methods.pushItem).toHaveBeenCalled();
  wrapper.find('[data-testid="one-unshiftItem"]').simulate('click');
  expect(methods.unshiftItem).toHaveBeenCalled();
  wrapper.find('[data-testid="one-removeItem-0"]').simulate('click');
  expect(methods.removeItem).toHaveBeenCalledWith(0);
  wrapper.find('[data-testid="one-moveItemUp-0"]').simulate('click');
  expect(methods.removeItem).toHaveBeenCalledWith(0);
  wrapper.find('[data-testid="one-moveItemDown-0"]').simulate('click');
  expect(methods.removeItem).toHaveBeenCalledWith(0);
});
