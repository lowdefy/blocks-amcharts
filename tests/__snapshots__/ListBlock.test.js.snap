// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render areas.content - value[0] 1`] = `
<div
  data-testid="areas.content"
  id="areas.content"
  style={
    Object {
      "border": "1px solid blue",
    }
  }
>
  <button
    data-testid="areas.content-pushItem"
    onClick={[MockFunction]}
  >
    pushItem
  </button>
  <div>
    <h4>
      Item Title
       
      0
      <div>
        <button
          data-testid="areas.content-removeItem-0"
          onClick={[Function]}
        >
          removeItem
        </button>
         
        <button
          data-testid="areas.content-moveItemUp-0"
          onClick={[Function]}
        >
          moveItemUp
        </button>
         
        <button
          data-testid="areas.content-moveItemDown-0"
          onClick={[Function]}
        >
          moveItemDown
        </button>
      </div>
    </h4>
    <div>
      <div
        style={
          Object {
            "border": "1px solid red",
            "padding": 10,
          }
        }
      >
        two
      </div>
    </div>
  </div>
  <div>
    <h4>
      Item Title
       
      1
      <div>
        <button
          data-testid="areas.content-removeItem-1"
          onClick={[Function]}
        >
          removeItem
        </button>
         
        <button
          data-testid="areas.content-moveItemUp-1"
          onClick={[Function]}
        >
          moveItemUp
        </button>
         
        <button
          data-testid="areas.content-moveItemDown-1"
          onClick={[Function]}
        >
          moveItemDown
        </button>
      </div>
    </h4>
    <div>
      <div
        style={
          Object {
            "border": "1px solid red",
            "padding": 10,
          }
        }
      >
        two
      </div>
    </div>
  </div>
  <button
    data-testid="areas.content-unshiftItem"
    onClick={[MockFunction]}
  >
    unshiftItem
  </button>
</div>
`;

exports[`Render blocks - value[0] 1`] = `
<div
  data-testid="blocks"
  id="blocks"
  style={
    Object {
      "border": "1px solid blue",
    }
  }
>
  <button
    data-testid="blocks-pushItem"
    onClick={[MockFunction]}
  >
    pushItem
  </button>
  <div>
    <h4>
      Item Title
       
      0
      <div>
        <button
          data-testid="blocks-removeItem-0"
          onClick={[Function]}
        >
          removeItem
        </button>
         
        <button
          data-testid="blocks-moveItemUp-0"
          onClick={[Function]}
        >
          moveItemUp
        </button>
         
        <button
          data-testid="blocks-moveItemDown-0"
          onClick={[Function]}
        >
          moveItemDown
        </button>
      </div>
    </h4>
    <div>
      <div
        style={
          Object {
            "border": "1px solid red",
            "padding": 10,
          }
        }
      >
        one
      </div>
    </div>
  </div>
  <button
    data-testid="blocks-unshiftItem"
    onClick={[MockFunction]}
  >
    unshiftItem
  </button>
</div>
`;

exports[`Render default - value[0] 1`] = `
<div
  data-testid="default"
  id="default"
  style={
    Object {
      "border": "1px solid blue",
    }
  }
>
  <button
    data-testid="default-pushItem"
    onClick={[MockFunction]}
  >
    pushItem
  </button>
  <button
    data-testid="default-unshiftItem"
    onClick={[MockFunction]}
  >
    unshiftItem
  </button>
</div>
`;

exports[`Render property.itemTitle - value[0] 1`] = `
<div
  data-testid="property.itemTitle"
  id="property.itemTitle"
  style={
    Object {
      "border": "1px solid blue",
    }
  }
>
  <button
    data-testid="property.itemTitle-pushItem"
    onClick={[MockFunction]}
  >
    pushItem
  </button>
  <div>
    <h4>
      Foo Item Title
       
      0
      <div>
        <button
          data-testid="property.itemTitle-removeItem-0"
          onClick={[Function]}
        >
          removeItem
        </button>
         
        <button
          data-testid="property.itemTitle-moveItemUp-0"
          onClick={[Function]}
        >
          moveItemUp
        </button>
         
        <button
          data-testid="property.itemTitle-moveItemDown-0"
          onClick={[Function]}
        >
          moveItemDown
        </button>
      </div>
    </h4>
    <div>
      <div
        style={
          Object {
            "border": "1px solid red",
            "padding": 10,
          }
        }
      >
        one
      </div>
    </div>
  </div>
  <button
    data-testid="property.itemTitle-unshiftItem"
    onClick={[MockFunction]}
  >
    unshiftItem
  </button>
</div>
`;

exports[`Test Schema areas.content 1`] = `true`;

exports[`Test Schema areas.content 2`] = `null`;

exports[`Test Schema blocks 1`] = `true`;

exports[`Test Schema blocks 2`] = `null`;

exports[`Test Schema default 1`] = `true`;

exports[`Test Schema default 2`] = `null`;

exports[`Test Schema property.itemTitle 1`] = `true`;

exports[`Test Schema property.itemTitle 2`] = `null`;
